### ISSUE TITLE
need zero_grad

### ISSUE THREAD

**Issue by K-Mike** [IDEA-opensource/DN-DETR#19]
https://github.com/IDEA-opensource/DN-DETR/blob/206fa267ba7df978fa968edda9f7dd351a4b72c1/engine.py#L84-L88

It looks like you forgot optimizer.zero_grad().

**Comment by K-Mike** [comment]
As far as I understand, new gradients are produced first, and then the clip comes, i.e. at each step you accumulate all gradients which were previously calculated.
